docs/naps/7-key-binding-dispatch.md

```diff
@@ -50,21 +52,30 @@ a **key binding** binds a key sequence to a command with conditional activation
 
 ### Key binding validity: convenience vs. complexity
 
-Some users want to use traditional modifier keys as a base key in key binding for convenience purposes [^id2]. However, this can lead to conflicts since many key bindings may include the modifier key in their key sequence and thus cause confusion and cost extra engineering effort.
+Some users want to use traditional modifier keys as a 'base key' in key binding for convenience purposes [^id2]. However, this can lead to conflicts since many key bindings may include the modifier key in their key sequence and thus cause confusion and cost extra engineering effort.
 
-The proposed restrictions on what key sequences can be used in a key binding aim to allow for the simplest user need while cutting down on any unnecessary complexities:
+We propose that:
 
-- key combinations containing any modifier keys as a base key are invalid
-- key chords cannot contain a base key that is a modifier key (aka a single modifier)
+- key combinations can either contain one modifier key (e.g., `alt`) or any number of
+  modifier keys plus a base key (e.g., `alt+shift+v`).
+  Multiple modifier keys (e.g., `alt+shift`) are not allowed.
+- a key chord part cannot be an invalid key combination nor a single modifier.
+    - `alt t` is **invalid** because the first part is a single modifier (even though
+      it is a valid key combination)
+    - `ctrl+x alt` is **invalid** because the second part is a single modifier
+    - `ctrl+x alt+v` is **valid**
+    - `meta meta` is **invalid** because both parts are single modifiers
 
-Here are some examples:
-- `alt` is **valid** as a base key because it contains no other modifiers and no other parts
-- `alt+meta` is **invalid** because it is a key combination comprised of only modifiers
-- `alt+t` is **valid** as a key combination
-- `alt t` is **invalid** because it is a key chord whose first part is a single modifier
-- `ctrl+x alt` is **invalid** because it is a key chord whose second part is a single modifier
-- `ctrl+x alt+v` is **valid** as a key chord
-- `meta meta` is **invalid** because it is a key chord comprised of only single modifier parts
+The proposal restricts modifier keys being used without a base key, except in the case
+of a single modifier key used in isolation.
+We decided to allow this as the use cases were compelling,
+for example binding `shift` to make a labels layer invisible during press and visible
+on release.
```

Comment:
Does this require changes in app-model? I thought it did but I might be misremembering.

Reply:
Nope! app-model is very flexible and considers almost any key sequence as valid.